Add template for Objective-C iOS framework with dependencies

DesignatedInitializer opened the pull request:
Adds a template for creating an Xcode project for an Objective-C framework that's already setup with:
・Framework target
・Test target
・Example app for testing integration and use of framework from an app
・Carthage and Cocoa Pods integration
・.gitignore file for common Objective-C ignores
・README, TODO and CHANGELOG templates

DesignatedInitializer commented on Objective-C iOS framework with dependencies/PROJECT_TEMPLATE/PRODUCT_PLACEHOLDER iOS Example/Sources/main.m:
```diff
@@ -0,0 +1,17 @@
+
+#import "AppDelegate.h"
+
+@import UIApplicationMain
```
This should be `UIKit`.

DesignatedInitializer commented on Objective-C iOS framework with dependencies/PROJECT_TEMPLATE/PRODUCT_PLACEHOLDER iOS Tests/Configuration/Test.debug.xcconfig:
```diff
@@ -0,0 +1,7 @@
+//  Test.debug.xcconfig
+//  Sunpo
+//
+//  Created by Alexander I. Kovács on 3/28/16.
```
Fix project name and author.

DesignatedInitializer commented on Objective-C iOS framework with dependencies/PROJECT_TEMPLATE/PRODUCT_PLACEHOLDER iOS/Configuration/App.debug.xcconfig:
```diff
@@ -0,0 +1,7 @@
+//
+//  Debug.xcconfig
```
Change to `App.debug.xconfig`.

DesignatedInitializer commented on Objective-C iOS framework with dependencies/PROJECT_TEMPLATE/PRODUCT_PLACEHOLDER iOS/Configuration/App.release.xcconfig:
```diff
@@ -0,0 +1,7 @@
+//
+//  Release.xcconfig
```
Change to `App.release.xconfig`.

DesignatedInitializer commented:
@weissazool, do you have time to eyeball this before I merge?  I've developed this for private use but we can use it as template for internal framework repositories as well as it'll save a ton of time setting up new projects.

weissazool commented:
Looking now.

weissazool commented on Objective-C iOS framework with dependencies/PROJECT_TEMPLATE/.gitignore:
```diff
@@ -0,0 +1,39 @@
+# Xcode
```
Why two .gitignore files?

DesignatedInitializer replied:
So the one in the root directory of the repository is the one actually used by this repository.  The second one here is the one that will be used by the repository created by the template.

weissazool replied:
Got it.

weissazool commented on Objective-C iOS framework with dependencies/PROJECT_TEMPLATE/.gitignore:
```diff
@@ -0,0 +1,39 @@
+*.mode2v3
+!default.mode2v3
+*.perspectivev3
+!default.perspectivev3
```
Any idea what these files are?

DesignatedInitializer replied:
No, this is from the template at https://github.com/github/gitignore/blob/master/Objective-C.gitignore

weissazool replied:
Should we be ignoring files that we don't actually know what they are?

DesignatedInitializer replied:
Are you asking me to question the combined wisdom of the folks that compiled that template?

weissazool replied:
Yup.

weissazool commented on Objective-C iOS framework with dependencies/PROJECT_TEMPLATE/Cartfile.resolved:
```diff
@@ -0,0 +1 @@
+github "CocoaLumberjack/CocoaLumberjack" "2.3.0"
```
Seems unnecessary to specify the version as part of the template.

DesignatedInitializer replied:
This file should've been ignored, I must have checked it in before adding the `.gitignore` file.  Will remove it.

DesignatedInitializer replied:
Not sure what you mean, it the `.resolved` file which should be part of the checked-in project, so it'll contain whatever Carthage puts into it.

weissazool commented on Objective-C iOS framework with dependencies/PROJECT_TEMPLATE/Podfile.lock:
```diff
@@ -0,0 +1,3 @@
+PODFILE CHECKSUM: c1ebee5d41ff4b455eb96960c52a66703463dddd
+
+COCOAPODS: 1.0.0
```
Similar to the Cartfile.resolved, I feel like the Podfile.lock will be automatically generated by install/bootstrap and is therefore unnecessary here.

DesignatedInitializer replied:
Oh, you mean ship the template without these two files and have the user updated dependencies and check the files in then?  That's not a bad idea, the instructions already say to run the script to update dependencies at least once, I'll add a comment about the `Cartfile.resolved` and `Podfile.lock`.

weissazool replied:
Yeah, my concern is that if you run `pod install` or `carthage bootstrap` those commands will just install/bootstrap what's listed in the `.resolved`/`.lock` files rather than checking for changes the user has made to the `Cartfile` or `Podfile`.

weissazool commented on Objective-C iOS framework with dependencies/PROJECT_TEMPLATE/README.md:
```diff
@@ -0,0 +1,32 @@
+
+## License
+
+Published under the [The MIT License (MIT)](LICENSE), see the LICENSE file in this repository for details.
```
As your personal template, this is fine, but I think we need to talk to Ed about what software license is appropriate for TenTen's private stuff (and TenTen's open source stuff if we release any).

DesignatedInitializer replied:
I was torn between adding a default license (which you'd then go and update when using the template) and leaving this empty to be filled in later.  I'll leave it as is for the moment, the whole idea being that this is a template and you'll always never use it exactly the way it is (but still save a ton of time for general setup).  In a future update, might have the license part be configurable as well.

weissazool replied:
👍

weissazool commented on README.md:
```diff
@@ -1 +1,24 @@
+## Overview
+
+This repository contains various templates for Xcode projects. They are already set up with
+・Various targets including the main app/framework target but also test targets
```
Various -> various
targets -> targets.

weissazool replied:
Never mind. Make that...
・Various -> - Various
etc.

weissazool commented on README.md:
```diff
@@ -1 +1,24 @@
+
+## Overview
+
+This repository contains various templates for Xcode projects. They are already set up with
```
with -> with:

weissazool commented:
Initial review done.

DesignatedInitializer commented:
Thanks for the review.  Addressed your comments and made some changes based on myself now creating a couple of projects with the template.  Most notably, I removed the Cocoa Pods integration as I usually prefer Carthage and ended up deleting the Cocoa Pods integration for most projects.  Might do a separate template instead that just has Cocoa Pods but no Carthage integration once I'm happy with this template.

weissazool commented:
One thought: Do you want to include a circle.yml file in the template?

weissazool commented:
(For CircleCI integration?)

DesignatedInitializer commented:
Hey, that was my idea when we talked about CircleCIlater in the afternoon!  How did you manage to backdate this comment?

weissazool commented:
I thought you had seen the comment and that was why you brought it up, heh.

DesignatedInitializer commented:
I made some final changes and squashed the commits together.  Also added the `circle.yml` file we talked about.  Will merge this now, this is a goof first version.
